ubuntu-advantage

```diff
@@ -313,10 +327,27 @@ is_livepatch_enabled() {
     canonical-livepatch status >/dev/null 2>&1
 }
 
-validate_user_pass_token(){
+validate_user_pass_token() {
     echo "$1" | grep -q '^[^:]\+:[^:]\+$'
 }
 
+check_credentials() {
+    local url
+    url=$(private_repo_url "$1" "$2")
+
+    case $(http_request_code "$url") in
+        200)
+            return 0
+            ;;
+        401)
+            error_msg 'Invalid credentials'
+            ;;
+        *)
+            error_msg 'Failed checking credentials'
```

Comment:
We should include the http code we got here in this message.

Reply:
done